File: OneQuestionPerDay/1/two_sum.go
package main

func twoSum(nums []int, target int) []int {
	cache := make(map[int]int, 0)
	for i, n := range nums {
		m := target - n
		if j, ok := cache[m]; ok {
			return []int{j, i}
		}
		cache[n] = i
	}
	return nil
}

File: OneQuestionPerDay/1/two_sum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        cache = {}
        for i, n in enumerate(nums):
            m = target - n
            if m in cache:
                return [cache[m], i]
            cache[n] = i
        return []